src/bootstrap/config.rs

```diff
@@ -1368,6 +1370,16 @@ impl Config {
         config.llvm_plugins = llvm_plugins.unwrap_or(false);
         config.rust_optimize = optimize.unwrap_or(true);
 
+        let fully_optimized = config.rust_optimize && debug != Some(true) && !config.incremental;
+        if fully_optimized && rust_codegen_units.is_none() {
+            rust_codegen_units = Some(1)
+        }
+        if fully_optimized && rust_codegen_units_std.is_none() {
+            rust_codegen_units_std = Some(1)
```

Comment:
this is a pre-existing bug and doesn't affect your pr, but i think this should default to `rust_codegen_units` if it's set, not have an independent default, like how `debug-assertions-std` works